sdk/ai/azure-ai-projects/samples/agents/sample_agents_basics_with_tracing.py

```diff
@@ -0,0 +1,84 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+FILE: sample_agents_basics_with_tracing.py
+
+DESCRIPTION:
+    This sample demonstrates how to use basic agent operations from
+    the Azure Agents service using a synchronous client with tracing.
+
+USAGE:
+    python sample_agents_basics_with_tracing.py
+
+    Before running the sample:
+
+    pip install azure.ai.projects azure-identity
+
+    Set this environment variables with your own values:
+    PROJECT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import os
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.projects.tracing import AIAgentsInstrumentor
+from tracing_helpers import configure_tracing
+# To use console exporter, uncomment following lines and install opentelemetry-sdk
+from opentelemetry import trace
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import SimpleSpanProcessor, ConsoleSpanExporter
+
+# Create an AI Project Client from a connection string, copied from your AI Studio project.
+# At the moment, it should be in the format "<HostName>;<AzureSubscriptionId>;<ResourceGroup>;<HubName>"
+# Customer needs to login to Azure subscription via Azure CLI and set the environment variables
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+
+scenario = os.path.basename(__file__)
+#tracer = configure_tracing("agent-samples").get_tracer(scenario)
+# To setup tracing to console, comment out the previous line and
+# uncomment the four lines below. Requires opentelemetry-sdk.
+exporter = ConsoleSpanExporter()
+trace.set_tracer_provider(TracerProvider())
+tracer = trace.get_tracer(__name__)
+trace.get_tracer_provider().add_span_processor(SimpleSpanProcessor(exporter))
+
+AIAgentsInstrumentor().instrument()
+
```

Comment:
As FYI, after I complete my next PR, instead of lines 43 to 52, this will be replaced by one call:
project_client.telemetry.enable(destination=sys.stdio)

Reply:
ok i can do another pr later to align

Reply:
changed